hrabsence/CRM/HRAbsence/Upgrader.php

```diff
@@ -551,4 +551,20 @@ public function upgrade_1401() {
 
     return TRUE;
   }
+
+  /**
+   * Previously when a leave type was deleted it was only deleting the first 25 civicrm_hrjobcontract_leave
+   * entries for that type. This upgrade is to remove legacy data where civicrm_hrjobcontract_leave points to a
+   * civicrm_hrabsence_type that does not exist
+   *
+   * @see https://compucorp.atlassian.net/browse/PCHR-1892
+   */
+  public function upgrade_1402() {
+    $deleteQuery = "DELETE FROM civicrm_hrjobcontract_leave WHERE leave_type NOT IN (
```

Comment:
@davialexandre  here he is talking to another extension (job contract extension) .. I know it will hurt the performance but this an upgrader that will be executed  only once .. so I suggest to use the API here .. and for absence types list , he can fetch them via BAO .